types/prosemirror-inputrules/index.d.ts

```diff
@@ -1,30 +1,111 @@
-// Type definitions for prosemirror-inputrules 0.21
+// Type definitions for prosemirror-inputrules 1.0
 // Project: https://github.com/ProseMirror/prosemirror-inputrules
 // Definitions by: Bradley Ayers <https://github.com/bradleyayers>
 //                 David Hahn <https://github.com/davidka>
+//                 Tim Baumann <https://github.com/timjb>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.1
 
-import { Node, NodeType } from 'prosemirror-model';
+// IMPORTANT
+// This file was generated by https://github.com/bradleyayers/getdocs2ts. Please do not edit manually.
+// When you find an error in these declarations, fix the getdocs comment upstream or 'getdocs2ts', then regenerate.
+
+import { Node as ProsemirrorNode, NodeType } from 'prosemirror-model';
 import { EditorState, Plugin, Transaction } from 'prosemirror-state';
 
+/**
+ * Input rules are regular expressions describing a piece of text
+ * that, when typed, causes something to happen. This might be
+ * changing two dashes into an emdash, wrapping a paragraph starting
+ * with `"> "` into a blockquote, or something entirely different.
+ */
 export class InputRule {
-  constructor(match: RegExp, handler: string | ((state: EditorState, match: string[], start: number, end: number) => Transaction | null | undefined))
+  /**
+   * Create an input rule. The rule applies when the user typed
+   * something and the text directly in front of the cursor matches
+   * `match`, which should probably end with `$`.
+   *
+   * The `handler` can be a string, in which case the matched text, or
+   * the first matched group in the regexp, is replaced by that
+   * string.
+   *
+   * Or a it can be a function, which will be called with the match
+   * array produced by
+   * [`RegExp.exec`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec),
+   * as well as the start and end of the matched range, and which can
+   * return a [transaction](#state.Transaction) that describes the
+   * rule's effect, or null to indicate the input was not handled.
+   */
+  constructor(match: RegExp, handler: string | ((state: EditorState, match: string[], start: number, end: number) => Transaction | null | void));
```

Comment:
Do you think we should include `undefined` in these return types (in addition to void)?

Reply:
The `void` type already includes the value `undefined`.